### ISSUE TITLE
Some errors in the resulting topologies after conversion to gmx

### ISSUE THREAD

**Issue by kaplajon** [alanwilter/acpype#24]
Hi,

I'm preparing a glycoprotein using tleap and glycam06j/ff14SB. Since I want to simulate in gromacs I used ACPYPE (the :latest docker image recipe built with Singularity last week). I also used tleap from the same image. Unfortunately I found some errors in the generated output that I had to fix before gromacs would run, that I thought you might want to know about.

1. In the CL- [moleculetype] the atom type is IM, but in [atomtypes] the type is Cl-. Changing IM to Cl- seems to be the fix for getting the topology through grompp.
2. In the output gro-file, the box vectors were written in Ångström, but the coordinates in nm. This made gromacs die with a segfault since it couldn't allocate enough memory for the resulting huge box. Dividing the box vectors by 10 fixed the segfault issue.

Thanks for a very nice python tool!
Cheers

**Comment by alanwilter** [comment]
Hi, thanks for using ACPYPE and for your report. About 1), can you please see #22. It's a similar issue which was addressed recently so you see it in the latests `acpype.py`.

I know a collaborator mentioned ACPYPE in a docker but where exactly did you get it? Thanks.

**Comment by kaplajon** [comment]
Hi, yes I saw that a similar issue with the sodium ion topology was fixed after I posted here. Thanks!

the docker image comes from the lpkagami/acpype:latest as mentioned in the [acpype.sh](https://github.com/alanwilter/acpype/blob/master/acpype.sh) from this repo. I thought it was the "official" one, but maybe I'm mistaken (I just wanted a quick way of using acpype together with ambertools)? Apart from the issues I've mentioned, it seems to work well for my purposes (amber2gmx functionality).

From the log file:
| ACPYPE: AnteChamber PYthon Parser interfacE v. 2020-07-25T09:06:13CEST (c) 2021 AWSdS

So it's a bit old it seems, maybe the issue has been fixed. I scanned the commits quickly, so I may have missed something important. :)

**Comment by alanwilter** [comment]
Hi, yes, it's our Docker, but we didn't advertise it yet because it not fully functional.

For using as "amber2gmx", just the acpype.py file should do the trick.

And yes, that version is a bit "old".

Nevertheless, I'm still puzzled with you topic (2) and I need to investigate the 10 off factor you spotted. Do you mind sending your input files so I can run some tests here? Thanks again.

**Comment by kaplajon** [comment]
Hi,

I'm not sure I'm allowed to share the topologies just yet, but I think it may be [this line](https://github.com/alanwilter/acpype/blob/ad9727c53234ae4266e9e453819459e7875a4c7f/acpype_lib/acpype.py#L4139) in acpype.py. I'm not adding solvent or setting any box with tleap, so acpype is guessing the box info from the coordinates and then multiplying with 20 it seems.

**Comment by alanwilter** [comment]
Yes, this is the bit of code that does is, I looked there as well and originally I applied our heuristic rules, besides, historically, [Amber](http://ambermd.org/Questions/units.html) always used Angstroms (0.1 nm) and Gromacs nm.

The thing here is, probably, my code is not addressing you scenario correctly because I used to work with PDB files, which uses Angs as well and usually have a PBC values.

I don't need your example anymore, I just created one, now I need a PDBs with different PBCs (rhombic, triclinic, octa etc.) test the "common" usage, working on it.

My guess is I just need to add the `amb2gmx` flag so multiply by 2 and 20 in this scenario.

**Comment by alanwilter** [comment]
Both problems are fixed on `main`. I re-ran an `amb2gmx` conversion of a solvated, ionised system (`RAMP1_ion.prmtop` / `.inpcrd`) to check each one.

**1. Ion atom type mismatch** — the `CL-` moleculetype now references the same type name that `[ atomtypes ]` declares, so `grompp` is happy:

```
[ atomtypes ]
 Na+      Na+      22.99000  0.00000   A   2.43928e-01  3.65846e-01
 Cl-      Cl-      35.45000  0.00000   A   4.47766e-01  1.48913e-01

[ moleculetype ]  CL-
; id_  at type  res nr  residue name  at name  cg nr  charge   mass
   1     Cl-      1        CL-          CL-      1     -1    35.45300
```

No `IM` anywhere.

**2. Box vectors in the wrong unit** — coordinates and box are both in nm now:

```
inpcrd box (Angstrom):  65.3823686 ...
gro box line (nm):       6.53824  6.16432  5.33845 ...
a coordinate line (nm):  1 CYX  N  1   4.231  2.692  4.833
```

65.38 Å becomes 6.538 nm, consistent with the coordinates, so no more segfault from an enormous box.

Sorry it took so long to confirm — and thank you for the precise report, which made both easy to verify.
